source/MaterialXGraphEditor/Graph.cpp

```diff
@@ -2655,59 +2660,63 @@ void Graph::deleteNode(UiNodePtr node)
             }
         }
     }
-    // update downNode info
-    std::vector<Pin> outputConnections = node->outputPins.front().getConnections();
 
-    for (Pin pin : outputConnections)
+    if (node->outputPins.size() > 0)
```

Comment:
This diff looks messy.. But the only change is this one line guard (and added indentation).